diff --git a/rtl/RamMLAB.vhd b/rtl/RamMLAB.vhd
--- a/rtl/RamMLAB.vhd
+++ b/rtl/RamMLAB.vhd
@@ -31,7 +31,7 @@ begin
       (
          indata_aclr                         => "OFF",
          indata_reg                          => "INCLOCK",
-         intended_device_family              => "Cyclone V",
+         intended_device_family              => "Agilex 5",
          lpm_type                            => "altdpram",
          outdata_aclr                        => "OFF",
          outdata_reg                         => "UNREGISTERED",
diff --git a/rtl/SyncRamDualByteEnable.vhd b/rtl/SyncRamDualByteEnable.vhd
--- a/rtl/SyncRamDualByteEnable.vhd
+++ b/rtl/SyncRamDualByteEnable.vhd
@@ -125,7 +125,7 @@ BEGIN
          clock_enable_output_a => "BYPASS",
          clock_enable_output_b => "BYPASS",
          indata_reg_b => "CLOCK1",
-         intended_device_family => "Cyclone V",
+         intended_device_family => "Agilex 5",
          lpm_type => "altsyncram",
          numwords_a => 2**ADDR_WIDTH,
          numwords_b => 2**ADDR_WIDTH,
diff --git a/rtl/dpram.vhd b/rtl/dpram.vhd
--- a/rtl/dpram.vhd
+++ b/rtl/dpram.vhd
@@ -38,7 +38,7 @@ begin
 		clock_enable_output_a => "BYPASS",
 		clock_enable_output_b => "BYPASS",
 		indata_reg_b => "CLOCK1",
-		intended_device_family => "Cyclone V",
+		intended_device_family => "Agilex 5",
 		lpm_type => "altsyncram",
 		numwords_a => 2**addr_width,
 		numwords_b => 2**addr_width,
@@ -136,7 +136,7 @@ begin
 		clock_enable_output_a => "BYPASS",                                      
 		clock_enable_output_b => "BYPASS",                                      
 		indata_reg_b => "CLOCK1",                                               
-		intended_device_family => "Cyclone V",                                  
+		intended_device_family => "Agilex 5",
 		lpm_type => "altsyncram",                                               
 		numwords_a => 2**addr_width_a,                                          
 		numwords_b => 2**addr_width_b,                                          
diff --git a/rtl/sdram.sv b/rtl/sdram.sv
--- a/rtl/sdram.sv
+++ b/rtl/sdram.sv
@@ -552,7 +552,7 @@ end
 altddio_out
 #(
 	.extend_oe_disable("OFF"),
-	.intended_device_family("Cyclone V"),
+	.intended_device_family("Agilex 5"),
 	.invert_output("OFF"),
 	.lpm_hint("UNUSED"),
 	.lpm_type("altddio_out"),
