diff --git a/Atari7800.sv b/Atari7800.sv
index 060e3d3..23cf25a 100644
--- a/Atari7800.sv
+++ b/Atari7800.sv
@@ -1,4 +1,4 @@
-`default_nettype none
+`default_nettype wire
 // k7800 (c) by Jamie Blanks
 
 // k7800 is licensed under a
@@ -11,6 +11,7 @@ module emu
 (
 	`include "sys/emu_ports.vh"
 );
+`default_nettype none
 
 ///////// Default values for ports not used in this core /////////
 
diff --git a/rtl/RIOT/M6532.sv b/rtl/RIOT/M6532.sv
index bf4f881..52c1f55 100644
--- a/rtl/RIOT/M6532.sv
+++ b/rtl/RIOT/M6532.sv
@@ -15,7 +15,7 @@ module M6532
 	input  [6:0] addr,      // Address
 	input        RW_n,      // 1 = read, 0 = write
 	input  [7:0] d_in,
-	output [7:0] d_out,
+	output logic [7:0] d_out,
 	input        RS_n,      // RAM select
 	output       IRQ_n,
 	input        CS1,       // Chip select 1, 1 = selected
@@ -161,4 +161,4 @@ end else begin
 	end
 end
 
-endmodule
\ No newline at end of file
+endmodule
diff --git a/rtl/TIA.sv b/rtl/TIA.sv
index 972257c..28ff8dc 100644
--- a/rtl/TIA.sv
+++ b/rtl/TIA.sv
@@ -1641,14 +1641,14 @@ module TIA
 	output logic    rdy,
 	input  [5:0]    addr,
 	input  [7:0]    d_in,
-	output [7:0]    d_out,
+	output logic [7:0] d_out,
 	input  [3:0]    i,     // On real hardware, these would be ADC pins. i0..3
-	output [3:0]    i_out,
+	output logic [3:0] i_out,
 	input           i4,
 	input           i5,
 	output [3:0]    aud0,
 	output [3:0]    aud1,
-	output [3:0]    col,
+	output logic [3:0] col,
 	output [2:0]    lum,
 	output          BLK_n,
 	output          sync,
@@ -1674,7 +1674,7 @@ module TIA
 	output          is_pal,
 	output          is_f1,
 	input           stabilize,
-	output  [3:0]   paddle_read // Helper to let us autodetect paddles
+	output logic [3:0] paddle_read // Helper to let us autodetect paddles
 );
 
 logic [7:0] wreg[64]; // Write registers. Only 44 are used.
@@ -2127,4 +2127,3 @@ always_ff @(posedge clk) begin : read_reg_block
 end
 
 endmodule
-
diff --git a/rtl/bram.vhd b/rtl/bram.vhd
index 83cb211..354ed83 100644
--- a/rtl/bram.vhd
+++ b/rtl/bram.vhd
@@ -29,8 +29,26 @@ END spram;
 
 
 ARCHITECTURE SYN OF spram IS
+	component spram_sz is
+		generic (
+			addr_width    : integer := 8;
+			data_width    : integer := 8;
+			numwords      : integer := 2**8;
+			mem_init_file : string := " ";
+			mem_name      : string := "MEM"
+		);
+		port (
+			clock   : in  std_logic;
+			address : in  std_logic_vector(addr_width-1 downto 0);
+			data    : in  std_logic_vector(data_width-1 downto 0) := (others => '0');
+			enable  : in  std_logic := '1';
+			wren    : in  std_logic := '0';
+			q       : out std_logic_vector(data_width-1 downto 0);
+			cs      : in  std_logic := '1'
+		);
+	end component;
 BEGIN
-	spram_sz : work.spram_sz
+	spram_sz_inst : spram_sz
 	generic map(addr_width, data_width, 2**addr_width, mem_init_file, mem_name)
 	port map(clock,address,data,enable,wren,q,cs);
 END SYN;
@@ -233,4 +251,3 @@ BEGIN
 	);
 
 END SYN;
-
diff --git a/rtl/cart2600.sv b/rtl/cart2600.sv
index 4ec7bc8..07fdd5a 100644
--- a/rtl/cart2600.sv
+++ b/rtl/cart2600.sv
@@ -5,11 +5,12 @@
 
 // You should have received a copy of the license along with this
 // work. If not, see http://creativecommons.org/licenses/by-nc/4.0/.
+`include "bankswitch_types.svh"
 
 module cart2600
 (
 	// Physical Pins
-	output   [7:0]  d_out, // Data bus
+	output logic [7:0] d_out, // Data bus
 	input    [7:0]  d_in,  // Data bus
 	input    [12:0] a_in,  // Address bus
 
diff --git a/rtl/detect2600.sv b/rtl/detect2600.sv
index 265f26e..d54b00f 100644
--- a/rtl/detect2600.sv
+++ b/rtl/detect2600.sv
@@ -1,9 +1,4 @@
-typedef enum bit[4:0] { 
-	BANK00, BANKF8, BANKF6, BANKFE, BANKE0,   BANK3F,   BANKF4,  BANKP2,
-	BANKFA, BANKCV, BANK2K, BANKUA, BANKE7,   BANKF0,   BANK32,  BANKAR,
-	BANK3E, BANKSB, BANKWD, BANKEF, BANKJANE, BANKDPCP, BANKCTY, BANKCDF,
-	BANKEND
-} bss_type ;
+`include "bankswitch_types.svh"
 
 module detect2600
 (
@@ -1175,4 +1170,3 @@ endmodule: match_bytes
 
 
 
-
diff --git a/rtl/paddles.sv b/rtl/paddles.sv
index be0275e..176dc33 100644
--- a/rtl/paddles.sv
+++ b/rtl/paddles.sv
@@ -10,7 +10,7 @@ module paddle_select
 (
 	input       clk,
 	input [7:0] value,
-	output      select
+	output logic select
 );
 	logic [7:0] last_value;
 	always_ff @(posedge clk) begin
@@ -185,7 +185,7 @@ module paddle_timer (
 	input [9:0] kohms,
 	input clear,
 	input read,
-	output charged,
+	output logic charged,
 	output logic [33:0] difference
 );
 
@@ -378,4 +378,4 @@ end
 // 	33'h112D2ABBC, 33'h11317D3B2, 33'h1135CFBA8, 33'h113A2239E, 33'h113E74B94, 33'h1142C738A, 33'h114719B80, 33'h114B6C376
 // };
 
-endmodule
\ No newline at end of file
+endmodule
diff --git a/rtl/top.sv b/rtl/top.sv
index 64481fb..81fd61d 100644
--- a/rtl/top.sv
+++ b/rtl/top.sv
@@ -578,7 +578,7 @@ module M6502C
 	output [15:0] AB,        // address bus
 	output [7:0]  DB_OUT,    // data_out,
 	output        RD,        // read enable
-	output        is_halted  // This is used to indicate that sally has released the bus
+	output logic  is_halted  // This is used to indicate that sally has released the bus
 );
 
 	logic cpu_halt_n = 1;
diff --git a/sys/video_mixer.sv b/sys/video_mixer.sv
index dc718df..aee2925 100644
--- a/sys/video_mixer.sv
+++ b/sys/video_mixer.sv
@@ -63,6 +63,10 @@ module video_mixer
 localparam DWIDTH_SD = GAMMA ? 7 : DWIDTH;
 localparam HALF_DEPTH_SD = GAMMA ? 0 : HALF_DEPTH;
 localparam RIN_DWIDTH = GAMMA ? 7 : DWIDTH;
+
+wire [RIN_DWIDTH:0] R_in;
+wire [RIN_DWIDTH:0] G_in;
+wire [RIN_DWIDTH:0] B_in;
 
 wire frz_hs, frz_vs;
 wire frz_hbl, frz_vbl;
@@ -91,13 +95,13 @@ end
 
 generate
 	if(GAMMA && HALF_DEPTH) begin
-		wire [7:0] R_in  = frz ? 8'd0 : {R,R};
-		wire [7:0] G_in  = frz ? 8'd0 : {G,G};
-		wire [7:0] B_in  = frz ? 8'd0 : {B,B};
-	end else begin
-		wire [DWIDTH:0] R_in = frz ? 1'd0 : R;
-		wire [DWIDTH:0] G_in = frz ? 1'd0 : G;
-		wire [DWIDTH:0] B_in = frz ? 1'd0 : B;
+		assign R_in = frz ? 8'd0 : {R,R};
+		assign G_in = frz ? 8'd0 : {G,G};
+		assign B_in = frz ? 8'd0 : {B,B};
+	end else begin
+		assign R_in = frz ? 1'd0 : R;
+		assign G_in = frz ? 1'd0 : G;
+		assign B_in = frz ? 1'd0 : B;
 	end
 endgenerate
 
