diff --git a/scaler.cpp b/scaler.cpp
index 96e7851..462625e 100644
--- a/scaler.cpp
+++ b/scaler.cpp
@@ -416,7 +416,7 @@ bool write_screenshot(const char *filename, const uint8_t *argb,
 static std::atomic<bool> screenshot_pending_atomic{false};
 static std::atomic<ScreenshotResult_atomic*> screenshot_result_data_atomic{nullptr};
 
-static struct { const char *fmtstr; Imlib_Load_Error errno; } err_strings[] = {
+static struct { const char *fmtstr; Imlib_Load_Error err_code; } err_strings[] = {
   {"file '%s' does not exist", IMLIB_LOAD_ERROR_FILE_DOES_NOT_EXIST},
   {"file '%s' is a directory", IMLIB_LOAD_ERROR_FILE_IS_DIRECTORY},
   {"permission denied to read file '%s'", IMLIB_LOAD_ERROR_PERMISSION_DENIED_TO_READ},
@@ -434,7 +434,7 @@ static struct { const char *fmtstr; Imlib_Load_Error errno; } err_strings[] = {
 static void print_imlib_load_error (Imlib_Load_Error err, const char *filepath) {
   int i;
   for (i = 0; err_strings[i].fmtstr != NULL; i++) {
-    if (err == err_strings[i].errno) {
+    if (err == err_strings[i].err_code) {
 	printf("Screenshot Error (%d): ",err);
 	printf(err_strings[i].fmtstr,filepath);
 	printf("\n");
diff --git a/user_io.cpp b/user_io.cpp
index 8e334c5..d754c9c 100644
--- a/user_io.cpp
+++ b/user_io.cpp
@@ -224,7 +224,7 @@ char is_menu()
 static int is_x86_type = 0;
 char is_x86()
 {
-	if (!is_x86_type) is_x86_type = strcasecmp(orig_name, "AO486") ? 2 : 1;
+	if (!is_x86_type) is_x86_type = (strcasecmp(orig_name, "AO486") && strcasecmp(orig_name, "PC110")) ? 2 : 1;
 	return (is_x86_type == 1);
 }
 
